CMakeLists.txt

```diff
@@ -80,14 +80,21 @@ set(TARGET_NAME ${COMMON_CLANG_LIBRARY_NAME}${BUILD_PLATFORM} )
 
 if(NOT USE_PREBUILT_LLVM)
     set(TARGET_BRANCH "ocl-open-90")
-    set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+
+    if(NOT LLVM_EXTERNAL_CLANG_SOURCE_DIR)
+      set(CLANG_SOURCE_DIR ${LLVM_SOURCE_DIR}/tools/clang)
+    elseif(EXISTS "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}/CMakeLists.txt") #monorepo
+      set(CLANG_SOURCE_DIR "${LLVM_EXTERNAL_CLANG_SOURCE_DIR}")
+    endif()
+    message(STATUS "CLANG_SOURCE_DIR = ${CLANG_SOURCE_DIR}")
```

Comment:
I think it is better to say something like "Using clang sources from /path/to/clang/sources".
My main concern is that `CLANG_SOURCE_DIR` looks like option name and someone might try to set it to override the path

Reply:
Done